src/cache/redis.ts

```diff
@@ -15,7 +15,34 @@ const logger = makeLogger('RedisCache')
 export class RedisCache<T = unknown> implements Cache<T> {
   type = CacheTypes.Redis
 
-  constructor(private client: Redis) {}
+  constructor(private client: Redis) {
+    this.loadFunctions()
+  }
+
+  async loadFunctions() {
+    // Load custom function 'setResponse' with lua script to redis that will skip overwriting successful cache response if the new value is error response
+    const source =
+      '#!lua name=ea \n ' +
+      "redis.register_function('setResponse', function(keys, args) " +
+      'local key = keys[1] ' +
+      'local value = args[1] ' +
+      'local ttl = tonumber(args[2]) ' +
+      'local json_value = cjson.decode(value) ' +
+      "local key_exists = redis.call('EXISTS', key) " +
+      'if json_value.errorMessage and key_exists == 1 then ' +
```

Comment:
This is not checking if the existing value in the cache is an error or not, so it would not overwrite the error with a more recent one. Was this done because of a performance limitation, or is it doable?

Reply:
I changed it to also check for stored value as well. My concern was performance since if the new value is error we need to GET the current value. On the other hand it all happens inside a function and should be very fast. Also with this we have the same logic as for local cache